src/main/services/csfloat/csfloatSocket.ts

```diff
@@ -0,0 +1,554 @@
+                    other_steam_id64: offer.partner.getSteamID64(),
+                } as OfferStatus
+            })
+
+            await this._csFloatClient.tradeOffers(offersForCSFloatMapped);
+        }
+
+        for (const offer of tradeOffersSents) {
+            const itemsToGive = offer.itemsToGive.map(
+                (item) => item.assetid
+            ) as string[];
+            const itemsToReceive = offer.itemsToReceive.map(
+                (item) => item.assetid
+            ) as string[];
+
+
+            if (offer.state !== ETradeOfferStateCSFloat.Active) continue;
+
+            const hasTradeWithNoOfferAnnotated = tradesInPending.find((trade) => {
+                if (trade.steam_offer.id) {
+                    return false;
+                }
+
+                return (itemsToGive || []).includes(trade.contract.item.asset_id);
+            });
+
+            if (!hasTradeWithNoOfferAnnotated) continue;
+
+            try {
+                await this._csFloatClient.annotateOffer({
+                    offer_id: offer.id,
+                    given_asset_ids: itemsToGive || [],
+                    received_asset_ids: itemsToReceive || [],
+                    other_steam_id64: offer.partner.getSteamID64(),
+                });
+            } catch (error) {
+                console.error(`failed to annotate offer ${offer.id} post-hoc`, error);
+            }
+        }
+    }
+
+    private async pingCancelTrades(
+        tradesInPending: ITradeFloat[],
+        tradeOffers: IGetTradeOffersResponse
+    ): Promise<void> {
+        const hasWaitForCancelPing = tradesInPending.find(
+            (trade) =>
+                trade.state === EStatusTradeCSFLOAT.PENDING &&
+                trade.wait_for_cancel_ping
+        );
+
+        if (!hasWaitForCancelPing) return;
+
+        const allTradeOffers = [
+            ...(tradeOffers.sent || []),
+            ...(tradeOffers.received || []),
+        ];
+
+        for (const trade of tradesInPending) {
+            if (trade.state !== EStatusTradeCSFLOAT.PENDING) continue;
+
+            if (!trade.wait_for_cancel_ping) continue;
+
+            const tradeOffer = allTradeOffers.find(
+                (steamTradeOffer) => steamTradeOffer.id === trade.steam_offer.id
+            );
+
+            if (
+                tradeOffer &&
+                (tradeOffer.state === ETradeOfferStateCSFloat.Active ||
+                    tradeOffer.state === ETradeOfferStateCSFloat.Accepted)
+            )
+                continue;
+
+            try {
+                await this._csFloatClient.pingCancelTrade({
+                    trade_id: trade.id,
+                    steam_id: this.steamIDBase64,
+                });
+            } catch (error) {
+                console.error(
+                    `failed to send cancel ping for trade ${trade.id}`,
+                    error
+                );
+            }
```

Comment:
Emit the error to be logged
```suggestion
            } catch (error) {
                console.error(
                    `failed to send cancel ping for trade ${trade.id}`,
                    error
                );
                this.emit("error", error)
            }
```